plugins/cache_promote/cache_promote.cc

```diff
@@ -110,7 +119,15 @@ TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
     return TS_ERROR;
   }
 
-  TSDebug(PLUGIN_NAME, "remap plugin is successfully initialized");
+  // Reserve a TXN slot for storing the calculated URL hash key
+  if (TS_SUCCESS != TSUserArgIndexNameLookup(TS_USER_ARGS_TXN, PLUGIN_NAME, &TXN_ARG_IDX, nullptr)) {
```

Comment:
Seems like this would always fail:
https://github.com/apache/trafficserver/blob/a0c267cdc88586679ccba4ec3e394f386d52f806/src/traffic_server/InkAPI.cc#L6330
Can't you just call reserve like a normal person?
https://github.com/apache/trafficserver/blob/a0c267cdc88586679ccba4ec3e394f386d52f806/plugins/esi/combo_handler.cc#L441